docs/runtime/runtime-metadata.md

```diff
@@ -0,0 +1,196 @@
+---
+title: "The Metadata of Runtime"
+---
+
+In blockchain, the logic to process transactions and modify storage is usually called STF (State Transition Function). The more popular name among Substrate developers is runtime. 
+
+Substrate module-based metadata provides an overview of blockchain runtime. It helps external code introspect modules, storage items, dispatchable functions and events.
+
+## SRML Metadata
+
+**[Srml-metadata](https://github.com/paritytech/substrate/tree/master/srml/metadata)** is not a "well-known" component of [SRML](runtime/substrate-runtime-module-library.md), since it doesn't provide additional storage entries, dispatchable functions and events. 
+
+It simply provides the data structure to store the overall runtime's metadata, and related codec to convert between raw data and bytes literal. Substrate framework uses the data format called SCALE (Simple Concatenated Aggregate Little-Endian), refer to [Low-level Data Format](overview/low-level-data-format.md) page for more information. 
+
+The structure of the runtime metadata could be changed in the future since Substrate framework is still in rapid evolvement. Here we are using `RuntimeMetadataV4` with Substrate v1.0 branch.
+
+```rust
+pub struct RuntimeMetadataV4 {
+    pub modules: DecodeDifferentArray<ModuleMetadata>,
+}
+```
+
+As you can see, the overall runtime's metadata is an array of module's metadata. These modules includes predefined SRML and your custom runtime modules. Each runtime module is mainly composed by storage entries, dispatchable functions and events.
+
+```rust
+pub struct ModuleMetadata {
+    pub name: DecodeDifferentStr,
+    pub prefix: DecodeDifferent<FnEncode<&'static str>, StringBuf>,
+    pub storage: ODFnA<StorageFunctionMetadata>,
+    pub calls: ODFnA<FunctionMetadata>,
+    pub event: ODFnA<EventMetadata>,
+}
+```
+
+Let's dig into the detail:
+* `name`: a string shows module's name.
+* `prefix`: an encodable function that produces store's metadata name.
+* `storage`: an array of storage entry's metadata which described in `StorageFunctionMetadata`. 
+* `calls`: an array of dispatchable function's metadata which described in `FunctionMetadata`.
+* `event`: an array of event's metadata which described in `EventMetadata`.
+
+```rust
```

Comment:
These code snippets don't really teach me anything and will probably immediately go out of sync with the real code.

Reply:
Fair point, I also think it's verbose, but could be helpful when the user digs into the codebase. I will definitely think about it and collect more feedback for this one.

Reply:
I also don't see any reason for showing this code. The user will never get into touch with this. When you want to write your own Metadata parser, you will have to look at the structures anyway. This copying of the structures does not bring anything.
as Joe already mentioned, this gets out of sync really fast.